### DISCUSSION TITLE
LinearFluidProperties Error

### DISCUSSION THREAD

**Discussion by Jake-HW-Miles** [idaholab/moose#28010] [Q&A Modules: Thermal Hydraulics]
Hi there,

I am experimenting with the thermal hydraulics functionality to hopefully do some system analysis. I have made an input file that seems to work with the fluid properties in simplefluidproperties, so wanted to implement more realistic properties using the linearfluidproperties instead.

The implementation is as follows:

```
[FluidProperties]
    # Placeholder fluid property
    [secondary_loop_fluid]
        type = LinearFluidProperties
        T_0 = ${temp_ref}
        p_0 = ${RTP_pressure}
        rho_0 = ${rho}
        cv = ${salt_specific_heat}
        a2 = 1e7 # Treat as incompressible hence dp/drho approaches infinity:
        beta = ${coeff_of_thermal_expansion}
        e_0 = 1.254e10
        k = ${k}
        mu = ${mu}
    []
[]
```

However am met with the following error:

```
*** ERROR ***
The following error occurred in the object "secondary_loop_fluid", of type "LinearFluidProperties".

secondary_loop_fluid: virtual void SinglePhaseFluidProperties::cv_from_v_e(libMesh::Real, libMesh::Real, libMesh::Real&, libMesh::Real&, libMesh::Real&) const derivatives not implemented.
```

At this stage, I require a work around - any suggestions or known fixes?

**Comment by GiudGiud** [comment]
Hello

The known fixes are:
- to implement this routine
- to use a different fluid property UO, for example SimpleFP (I saw you used that already) or one of the generic ones, or one of the specific ones if your field is in the list of the pre-implemented ones already. Which fluid are you using?

How comfortable are you with C++? For implementing

Guillaume

**Reply by GiudGiud** [reply]
Hello

I ended up taking a look and finding some mistakes in LinearFP.
The missing routine and the routine you needed can be found in this pull request
https://github.com/idaholab/moose/pull/28026

Guillaume

**Reply by Jake-HW-Miles** [reply]
Thank you very much Guillaume! I will keep an eye out for this

**Reply by joshuahansel** [reply]
Hi Jake, I'm also interested in which fluid you're using. You say you want to use a more realistic equation of state, but `LinearFluidProperties` might not fit that description. We've considered removing this class.

**Reply by Jake-HW-Miles** [reply]
I was originally going to use [NaClFluidProperties](https://mooseframework.inl.gov/source/fluidproperties/NaClFluidProperties.html)

But the range of validity concerned me, as I want liquid phase, so decided to define it another way. This does not need to perfect at this stage.

**Reply by GiudGiud** [reply]
Yes NaClFluidProperties is not designed for THM or the NS module. It s for porous flow applications

**Reply by Jake-HW-Miles** [reply]
Hi both! I see that #28026 has been merged now! Thank you, this is good to have.

The desire to use this FP in the first instance was to get some variation in the FP with temperature. For achieving a more realistic equation of state, I considered using the tabulated approach or [TemperaturePressureFunctionFluidProperties](https://mooseframework.inl.gov/source/fluidproperties/TemperaturePressureFunctionFluidProperties.html). Are both of these available? Should I prioritise one over the other?

**Reply by GiudGiud** [reply]
Hello

Both TabulatedFluidProperties and PressureTemperature...Fp are available 

Please let me if there are missing properties for either for your application if you try them

Guillaume
